core/src/components/datetime/datetime.tsx

```diff
@@ -1898,6 +1902,8 @@ export class Datetime implements ComponentInterface {
 
     return (
       <div
+        // Non-visible months should be hidden from screen readers
+        aria-hidden={!isWorkingMonth ? 'true' : 'false'}
```

Comment:
Hiding the months outside of the active working month, results in the screen reader not announcing the previous and next month days when the group is announced.